src/main/java/com/google/gwt/site/webapp/client/GWTProjectEntryPoint.java

```diff
@@ -443,23 +492,37 @@ private boolean isHomePage(String path) {
   }
 
   private void styleHomepage() {
-    final int windowHeight = $(window).height();
-    int sectionHeight = $("#letsbegin").height();
+    final int windowHeight = window.innerHeight;
+    int sectionHeight = Js.<HTMLElement>uncheckedCast(document.querySelector("#letsbegin")).offsetHeight;
 
     if (windowHeight > sectionHeight) {
-      $(".home section").each(new Function() {
-        @Override
-        public void f() {
-          $(this)
-              .css("height", windowHeight + "px")
-              .css("padding", "0");
-          GQuery container = $(this).find(".container");
-          container.css("padding-top", (windowHeight - container.height()) / 2 + "px");
-        }
+      forEach(".home section", element -> {
+        element.style.setProperty("height", windowHeight + "px");
+        element.style.setProperty("padding", "0");
+        HTMLElement container = Js.uncheckedCast(element.querySelector(".container"));
```

Comment:
For my part, I would generally prefer a real cast (e.g. `(HTMLElement) element.querySelector(...)`) for each of these rather than uncheckedCast - like `getElementOffset()` above does. It has the advantage of failing if the code doesn't work, it looks more like real java, and you can always compile it out if you hate time spent validating for ClassCastException.

Expert users like you know the difference and when it matters, but newcomers generally should see the less-unsafe way to do things - and if we actually need an uncheckedCast (because browser runtimes don't support their own types consistently...), we should have a comment for each unsafe thing.

Reply:
Changed everything to real casts.